### DISCUSSION TITLE
Requirements and best practices for updating minor and patch versions

### DISCUSSION THREAD

**Discussion by jludwiggreenaction** [trpc/trpc#4811] [Q&A] [answered]
Since updates occur in both the server and client separately, I am hoping to get some guidelines on updating this package:

- For minor and patch versions, is it safe to update one package at a time (e.g. would `@trpc/server` version 10.38.2 expected to be compatible with `@trpc/client` version 10.37.8)?
- If there is no compatibility guaranteed, are there any guidelines or other recommendations you have for updating?

Thanks.

**Comment by KATT** [comment]
On my phone but use something like `npx npm-check-updates /trpc/ -u` if I remember it correctly without looking it up


You're never safe to only update one of them without updating all. You'll also get peer dependency errors

**Reply by jludwiggreenaction** [reply]
Thanks for the response. And sorry for my super late reply but I meant to follow up with one additional question:

Running `npx npm-check-updates /trpc/ -u` makes sense for updating the packages. 

However, what about the compatibility between a server deployment and client deployment, e.g. trpc server running in a container and then a separate web SPA? Is it guaranteed that API between will remain the same between minor/patch versions?

**Reply by KATT** [reply, marked as answer]
A change in the RPC spec would be seen as a major bump
